test/shared/src/main/scala/zio/test/Spec.scala

```diff
@@ -218,6 +218,24 @@ final case class Spec[-R, +E, +L, +T](caseValue: SpecCase[R, E, L, T, Spec[R, E,
       case t @ TestCase(_, _)            => Spec(f(t))
     }
 
+  final def filter[R1 <: R, E1 >: E, L1 >: L, T1 >: T](
```

Comment:
Type annotations look good. Can we get `f` to return a `Boolean` instead of an `Option`?

Reply:
Yeah, that's a good point. You'd expect `Boolean` when encountering a `filter`. There wasn't a need to use `Option` there.

Reply:
Ooh, that's much nicer. Thanks! I applied your suggestions.

I'd like to add the suite filter too, but if you want that as part of this PR I think it's going to take me some time. I need to think about it more. It seems like to do it properly you'd need to do a depth-first search while maintaining history of the nodes you've visited. But maybe I'm over-complicating it. I haven't given it too much thought yet.

Reply:
No I definitely don't think we need to do the suite filter as part of this PR. I know you had mentioned it above and just trying to think ahead a bit but I agree it has some additional complications and I have a feeling this will be the more commonly used feature.